source/kraken/apply_disruption.cpp

```diff
@@ -54,6 +54,47 @@ namespace bg = boost::gregorian;
 
 namespace {
 
+static nt::VehicleJourney* create_vj_from_old_vj(nt::MetaVehicleJourney* mvj,
+                              const nt::VehicleJourney* vj,
+                              const std::string& new_vj_uri,
+                              nt::RTLevel rt_level,
+                              nt::ValidityPattern new_vp,
+                              const std::vector<nt::StopTime>& new_stop_times,
+                              nt::PT_Data& pt_data) {
+    auto* company = vj->company;
+    auto vehicle_journey_type = vj->vehicle_journey_type;
+    auto odt_message = vj->odt_message;
+    auto vehicle_properties = vj->_vehicle_properties;
+
+    auto* new_vj = mvj->create_discrete_vj(new_vj_uri,
+            rt_level,
+            new_vp,
+            vj->route,
+            std::move(new_stop_times),
+            pt_data);
+    vj = nullptr;
```

Comment:
why?

Reply:
`create_discrete_vj()` also triger the deletion of useless vjs, so when a disruption is applied on an already adapte/realtime vj, the *base* vj is deleted when we create the new one.
To be safe I set `vj = nullptr` to explicitly indicate that we can't use it anymore.

Reply:
A comment explaining why it is set to null would be great.